### ISSUE TITLE
Can't do anything...random error?

### ISSUE THREAD

**Issue by AeraWolfsRain** [textadventures/quest#930]
Hello, I was trying to do a ShowMenu and then switch with the results when everything just....somehow...broke? 

The error is: 
System.Exception: Starting transaction when previous transaction not finished
   at TextAdventures.Quest.UndoLogger.StartTransaction(String command)
   at TextAdventures.Quest.EditorController.DeleteElement(String elementKey, Boolean useTransaction)
   at TextAdventures.Quest.Editor.Delete()
   at TextAdventures.Quest.MainToolbar.HandleClick(String tag)
   at TextAdventures.Quest.MainToolbar.HandleClick(Object sender, EventArgs e)
   at System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
   at System.Windows.Forms.ToolStripButton.OnClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
   at System.Windows.Forms.ToolStripItem.FireEventInteractive(EventArgs e, ToolStripItemEventType met)
   at System.Windows.Forms.ToolStripItem.FireEvent(EventArgs e, ToolStripItemEventType met)
   at System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ScrollableControl.WndProc(Message& m)
   at System.Windows.Forms.ToolStrip.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

**Comment by ThePix** [comment]
Can you paste the code that caused it into a comment here? Or even better attach the whole game (see instructions at the bottom of the comment box), and tell me how to reproduce the error. What version of Windows and Quest are you using?

**Comment by kammererj2016** [comment]
I'm having a similar issue. Restarting fixes it but loses any unsaved work, which is a pain. As far as I can tell, it only happens when using First Time in the script editor. Below is the error code. I would attach the game, but Quest files aren't supported. For what it's worth, my error code looks more like the OP's the second time I try to do anything.

System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
   at TextAdventures.Quest.EditableScripts.Remove(Int32[] indexes)
   at TextAdventures.Quest.EditorControls.ScriptEditorControl.ctlToolbar_Delete()
   at TextAdventures.Quest.EditorControls.ScriptToolbar.cmdDelete_Click(Object sender, RoutedEventArgs e)
   at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.UIElement.RaiseEvent(RoutedEventArgs e)
   at System.Windows.Controls.Primitives.ButtonBase.OnClick()
   at System.Windows.Controls.Button.OnClick()
   at System.Windows.Controls.Primitives.ButtonBase.OnMouseLeftButtonUp(MouseButtonEventArgs e)
   at System.Windows.UIElement.OnMouseLeftButtonUpThunk(Object sender, MouseButtonEventArgs e)
   at System.Windows.Input.MouseButtonEventArgs.InvokeEventHandler(Delegate genericHandler, Object genericTarget)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.ReRaiseEventAs(DependencyObject sender, RoutedEventArgs args, RoutedEvent newEvent)
   at System.Windows.UIElement.OnMouseUpThunk(Object sender, MouseButtonEventArgs e)
   at System.Windows.Input.MouseButtonEventArgs.InvokeEventHandler(Delegate genericHandler, Object genericTarget)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
   at System.Windows.UIElement.RaiseEvent(RoutedEventArgs args, Boolean trusted)
   at System.Windows.Input.InputManager.ProcessStagingArea()
   at System.Windows.Input.InputManager.ProcessInput(InputEventArgs input)
   at System.Windows.Input.InputProviderSite.ReportInput(InputReport inputReport)
   at System.Windows.Interop.HwndMouseInputProvider.ReportInput(IntPtr hwnd, InputMode mode, Int32 timestamp, RawMouseActions actions, Int32 x, Int32 y, Int32 wheel)
   at System.Windows.Interop.HwndMouseInputProvider.FilterMessage(IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Interop.HwndSource.InputFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

**Comment by ThePix** [comment]
Zip the game file, and you should then be able to upload it.

When does it happen? When you load the game, start the game, do something in he editor, do something in the game, etc.? Does it crash Quest?

**Comment by kammererj2016** [comment]
Sorry, I just saw this. As a result of this error, Quest does not allow me to save the game, so I can't send a copy of the game in an error state, but I can reliably reproduce this error in any file, even a brand new one with nothing else in it, just by using firsttime and then trying to do anything else. I can look at other objects, rooms, etc, but the second I try to edit anything, I get an error thrown. It doesn't crash Quest, but it renders the file being worked on entirely unusable. You can still load a file, exit Quest, etc, but something about firsttime just corrupts a game.

**Comment by kammererj2016** [comment]
Note: It seems like writing the code out rather than using the UI will work without an error. Maybe it has something to do with an empty "otherwise"?

**Comment by ThePix** [comment]
I have had a play around and cannot reproduce. Using code this is fine:
```
firsttime {
  msg("First")
}
```
Tried this:
```
firsttime {
  msg("First")
}
otherwise
```
The editor converts it to this which is fine:
```
firsttime {
  msg("First")
}
otherwise {
}
```
So then tried adding bits in the GUI, going to an object and editing its description between adding each bit. First I put in a `firsttime`, then added a `Print` after it, then a `Print` for the `firsttime`, then a `Print` for the otherwise. I ended up with this in code, and again, all fine:
```
firsttime {
  msg ("First time only")
}
otherwise {
  msg ("Other times")
}
msg ("Here anyway")
